src/Source.php

```diff
@@ -27,14 +27,14 @@ final class Source
      * @param string $name The template logical name
      * @param string $path The filesystem path of the template if any
      */
-    public function __construct($code, $name, $path = '')
+    public function __construct(string $code, $name, string $path = '')
```

Comment:
why not adding the type on the name ?

Reply:
test did not pass and I hadn't time to have a look, fixed now.